test/EndToEnd/Tests/Framework/Script.cs

```diff
@@ -0,0 +1,53 @@
+namespace EndToEndTests.Framework
+{
+    using System.Diagnostics;
+
+    public class Script
+    {
+        /// <summary>
+        /// Represents the result of invoking a bash script.
+        /// </summary>
+        public class InvokeScriptResult
+        {
+            public int ExitCode { get; set; }
+            public string? StdOut { get; set; }
+            public string? StdErr { get; set; }
+        }
+
+        /// <summary>
+        /// Invokes a bash script on the local machine and returns InvokeScriptResult.
+        /// </summary>
+        public static InvokeScriptResult InvokeScript(string scriptName, string[] args)
+        {
+            var process = new Process
```

Comment:
Can we put all these codes inside using statement to ensure proper disposal even in the case of an exception.

Reply:
Is it required after having a finally block?

Reply:
Its more of a practice thing, even if we don't use using here, its fine.

Reply:
I have updated it!